docs/DeveloperGuide.md

```diff
@@ -1,34 +1,251 @@
+
+## 4. Implementation
+
+This section describes how some of the features in Fitr are implemented.
+
+### 4.1 Edit command
+
+When the user enters an edit command, it first passes through `Parser`. Once the input is parsed as an edit command, it is then passed to `EditCommandParser`, where it further parses the user's input. As the user is able to edit either the individual profile characteristics, food or exercise entries, the `EditCommandParser` is able parse what the user intends to edit. For example, if the user intends to edit a food entry, the input is then passed to `EditEntryCommand`, which parses the remaining arguments, and performs the required edit.
+Figure 4 below shows the sequence diagram when the user enters `edit exercise 25/10/2020 1 push ups /100`.
+
+![Sequence diagram for editing exercise](images/EditExerciseSequenceDiagram.png)
+
+_Figure 4: Sequence diagram for `edit` command_
+
+### 4.2 Help command
+
+When the user enters 'help', the `Ui` class reads it and passes it through `Parser`.  Once the input is parsed as a help command, the user input is passed to `HelpCommand`, which calls `Ui` to print the help message.
+Figure 5 below shows the sequence diagram when the user enters `help`.
+
+![Sequence diagram for help command](images/HelpCommandSequenceDiagram.png)
+
+_Figure 5: Sequence diagram for `help` command_
+
+### 4.3 View command
+
+When the user enters a view command, the `Ui` class reads it and passes it through `Parser`. Once the input is parsed as a view command, it is then passed to `ViewCommand`, where it is handled based on the type of view command. Then the respective view method is called to output the messages via the `Ui` class. For example, if you intend to view your goal entry, the input is passed to `ViewCommand`, which checks the arguments after 'view' and calls the `viewGoal()` method. It then performs the required steps to retrieve the goal status for each entry and prints the results using the `printCustomMessage()` method in the `Ui` class.
+
+![Sequence diagram for view goal command](images/ViewGoalSequenceDiagram.png)
+
+_Figure 6: Sequence diagram for `view goal` command_
+
+### 4.4 Clear command
+
+The `clear` command allows the user to clear either the exercise list, food list, or goal list, by specifying in the user's input as an argument (i.e. `clear exercise`, `clear food`, or `clear goal`). The user is also able to clear all the lists at the same time, if no argument is specified (i.e. `clear`).
+
+The user's input is first parsed by the `Parser` class. It is then passed to the `ClearCommand` class, which is then executed. The arguments are parsed in the `ClearCommand` class, and the required list(s) is then cleared. After clearing, it then writes the new empty list(s) to local storage.
+
+Figure 7 below shows the sequence diagram when the user inputs the `clear` command.
+
+![Sequence diagram for clear command](images/ClearCommandSequenceDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/57383789/97661912-a0d7de00-1ab0-11eb-9756-6e1f6fac8289.png)

Does view command access the variables of other objects, maybe it is possible to add them in.